### ISSUE TITLE
Pallet does not automatically insert packages installed for dependencies

### ISSUE THREAD

**Issue by sambrightman** [rdallasgray/pallet#38]
If a package is installed with dependencies, it is automatically inserted into `Cask` but the dependencies are not until some point later on (normally after `pallet-init` but perhaps not after the first one). I can't tell if this is intentional or not or what eventually causes them to be added.

**Comment by sambrightman** [comment]
I'm using pallet-init as "sync" to try to see if it helps. Normally I would only run it once (if at all).

My current situation, having installed [flx-ido](http://melpa.org/#/flx-ido) (first-level dependency on flx):

```bash
sam@wildthing:~$ ls -d .emacs.d/.cask/24.3.1/elpa/flx*
.emacs.d/.cask/24.3.1/elpa/flx-20140921.739  .emacs.d/.cask/24.3.1/elpa/flx-ido-20140821.2033
sam@wildthing:~$ grep flx ~/.emacs.d/Cask
(depends-on "flx-ido")
sam@wildthing:~$
```

I believe that just after filing this issue I re-ran pallet-init and this did not solve the problem. I've now installed an unrelated package (flx still missing, new package present) and then re-run pallet-init again. Now flx is listed.

I'm not really sure what the current expected behaviour is but it seems inconsistent/buggy. Am I wrong to expect flx to be there? I agree that top-level-only could be desirable in future (to stop stale dependencies persisting to new installs).

**Comment by sambrightman** [comment]
I agree that the normal way of running things would be to run pallet-init once and then never add lower-level dependencies after that. However, this does still mean that the initial setup can be overly broad (well, more like overly deep).

I'm currently testing a change which will only pallet-init packages that are not dependencies of other packages. Additionally, there is a second, heuristic part that can still include packages which appear to be manually installed. This gives me quite a clean pallet-init with only a couple of packages which I'd want to be listed/unlisted differently instead of a dozen. Ignored packages are always installed via dependencies anyway, even if they are not listed in the Cask file by pallet-init - so completeness should not be sacrificed.

Would you be open to such functionality, for example via pallet-init with prefix command?

**Comment by rdallasgray** [comment]
Yes -- in fact, it may be that this should be the default functionality. But starting with pallet-init via prefix command might be best.
